src/app/sc2/domain/slack.py

```diff
@@ -0,0 +1,115 @@
+""" Slack integration """
+import urllib
+
+from google.appengine.api import urlfetch
+
+from app.sc2.utils.match_utils import get_vs_string_from_match
+from app.sc2.utils.player_utils import prettify_name
+
+
+# TODO:slack hook this up when auth stuff figured out
+def update_channel_topic_with_open_games(channel_id, open_matches):
+    """
+    Sets the channel topic to the currently open games
+    Need to wait until I check out the auth stuff before this will work
+    """
+    if open_matches:
+        games_to_be_played = '\n'.join([get_vs_string_from_match(m) for m in open_matches])
+    else:
+        games_to_be_played = 'All games have been played! :boom:'
+
+    # TODO: get auth token for slack
+    payload = urllib.urlencode({
+        'token': 'asdf',
+        'channel': channel_id,
+        'purpose': games_to_be_played,
+    })
+
+    headers = {
+        'Content-type': 'application/x-www-form-urlencoded',
+    }
+
+    urlfetch.fetch(
+        'https://slack.com/api/channels.setPurpose',
+        payload=payload,
+        method=urlfetch.POST,
+        headers=headers
+    )
+
+
+def alert_match_closed(match):
+    """
+    Send a message to slack when a match is closed
+    :param match: the match that was closed
+    """
+    headers = {
+        'Content-type': 'application/x-www-form-urlencoded',
+    }
+    winner, loser = determine_match_winner_loser(match)
+    match_string = get_vs_string_from_match(match)
```

Comment:
Ya, I guess I could pass the match further down. Move the winner/loser stuff into there too